src/test/java/javatests/UserAuthUtilTest.java

```diff
@@ -0,0 +1,89 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package javatests;
+
+import java.util.*;
+
+import com.google.appengine.api.users.User;
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import org.apache.http.auth.AUTH;
+import org.junit.After;
+import org.junit.Assert.*;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import util.UserAuthUtil;
+import com.google.appengine.tools.development.testing.LocalServiceTestHelper;
+import com.google.appengine.tools.development.testing.LocalUserServiceTestConfig;
+
+import java.security.GeneralSecurityException;
+import java.io.IOException;
+
+import static org.junit.Assert.*;
+
+
+/** Test the functionality of the SheetsConverter class. */
+@RunWith(JUnit4.class)
+public final class UserAuthUtilTest {
+
+  private static final String UNAUTHORIZED_EMAIL = "example@foobar.com";
+  private static final String UNAUTHORIZED_DOMAIN = "foobar.com";
+  private static final String AUTHORIZED_EMAIL = "example@google.com";
+  private static final String AUTHORIZED_DOMAIN = "google.com";
+  private static final String CONSTRUCTOR_DOMAIN = "gmail.com";
+
+  /* Allows local unit testing of cloud API
+   * https://cloud.google.com/appengine/docs/standard/java/tools/localunittesting
```

Comment:
This seems pretty cool. How did you find out about this?

Reply:
Lots of googling 😎 , but I got this straight from the example in the API linked in that comment. I needed to be able to manually set user details with the API without making calls to the service--it's basically just mocks what the UserService does